### PR TITLE
Do not create transactions with incorrect claims

### PR DESCRIPTION
**Pull request by bvbfan**
Signed-off-by: Anthony Fieroni <bvbfan@abv.bg>

### CODE REVIEW

**Comment by bvbfan** [comment]
https://github.com/lbryio/lbrycrd/issues/116

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -426,6 +427,14 @@ UniValue claimname(const UniValue& params, bool fHelp)
         );
     string sName = params[0].get_str();
     string sValue = params[1].get_str();
+
+    if (sName.empty())
+        throw JSONRPCError(RPC_WALLET_ERROR, "Claim name should not be empty");
+
```
Comment (kaykurokawa) [inline src/wallet/rpcwallet.cpp]: while it is unintuitive, an empty claim name is actually valid (you can make claim for empty names when you use double quotes: "" in the RPC claimname command) so this check is not needed

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -847,7 +864,23 @@ UniValue supportclaim(const UniValue& params, bool fHelp)
         );
 
     string sName = params[0].get_str();
+
+    if (sName.empty())
+        throw JSONRPCError(RPC_WALLET_ERROR, "Claim name should not be empty");
+
```
Comment (kaykurokawa) [inline src/wallet/rpcwallet.cpp]: same here

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -847,7 +864,23 @@ UniValue supportclaim(const UniValue& params, bool fHelp)
...
+        throw JSONRPCError(RPC_WALLET_ERROR, "Claim id should be exact 160 symbols");
+
+    size_t i = 0;
+    if (sClaimId[0] == 0 && tolower(sClaimId[1]) == 'x')
+        i = 2;
+
+    for (; i < sClaimId.size(); ++i)
+        if (!isxdigit(sClaimId[i]))
+            throw JSONRPCError(RPC_WALLET_ERROR, "Claim id should contains only hexdecimal numbers");
+
```
Comment (kaykurokawa) [inline src/wallet/rpcwallet.cpp]: There is actually a function created to check claimId here: 
https://github.com/lbryio/lbrycrd/blob/master/src/rpc/claimtrie.cpp#L10

**Comment by bvbfan** [comment]
rpc/claimtrie.cpp is a part of libbitcoin_server while rpcwallet of libbitcoin_wallet that's why i'm including only 2 checks from ParseClaimtrieId not compete one.

**Comment by bvbfan** [comment]
Fix addressed issues.

**Comment by bvbfan** [comment]
Mac build looks broken again.

**Comment by BrannonKing** [comment]
Rebase off master again and the OSX build will be fixed.

**Comment by bvbfan** [comment]
Done

**Pull request merged**